### PR TITLE
Retry for TFTP data transmission

### PR DESCRIPTION
**Pull request by LipuFei**
TFTP retry mechanism for data transmission ONLY. Negotiation packets such as OP_RRQ and OP_OACK will not be retransmitted.

Closes #1392 and #903

### CODE REVIEW

**Comment by LipuFei** [comment]
retest this please

**Comment by NielsZeilemaker** [comment]
@LipuFei is this ready? / should we close #1392?

**Comment by LipuFei** [comment]
@NielsZeilemaker This one is ready. Feel free to close #1392 if you think this one is ok.

File: Tribler/Core/TFTP/handler.py [hunk 1]
```diff
@@ -454,6 +476,12 @@ def _handle_packet_as_receiver(self, session, packet):
         self._logger.debug(u"%s Got data, #block = %s size = %s", session, packet['block_number'], len(packet['data']))
 
         # check block_number
+        # ignore old ones, they may be retransmissions
+        if packet['block_number'] < session.block_number:
```
Comment (NielsZeilemaker) [inline Tribler/Core/TFTP/handler.py]: What's happening here? Are you enforcing the packets such that they arrive in order? You know that's not how the internet works right?

**Comment by NielsZeilemaker** [comment]
Seems if fine the comment I made "You only request one block at a time right?" is true.
If so, merge this one and close the other

File: Tribler/Core/TFTP/handler.py (hunk 1 above)
Comment (LipuFei) [inline Tribler/Core/TFTP/handler.py]: I asked @whirm, it's UDP so I am not sure if those packets will arrive in the same order...

**Comment by LipuFei** [comment]
@NielsZeilemaker It only requests one block at a time right. I will have another check later and close the other one. Thanks!

**Comment by LipuFei** [comment]
@whirm ready

**Pull request merged**